sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/ClientRetryPolicy.java

```diff
@@ -127,17 +128,22 @@ private ShouldRetryResult shouldRetryOnSessionNotAvailable() {
                 if (this.sessionTokenRetryCount > endpoints.size()) {
                     // When use multiple write locations is true and the request has been tried
                     // on all locations, then don't retry the request
+
+                    logger.warn("SessionNotAvailable: canUseMultipleWriteLocations = true, sessionTokenRetryCount = {}, retried all locations. retry exhausted!", sessionTokenRetryCount);
                     return ShouldRetryResult.noRetry();
                 } else {
+                    logger.info("SessionNotAvailable: canUseMultipleWriteLocations = true, sessionTokenRetryCount = {}, going to retry on next location", sessionTokenRetryCount);
                     this.retryContext = new RetryContext(this.sessionTokenRetryCount - 1, this.sessionTokenRetryCount > 1);
                     return ShouldRetryResult.retryAfter(Duration.ZERO);
                 }
             } else {
                 if (this.sessionTokenRetryCount > 1) {
                     // When cannot use multiple write locations, then don't retry the request if
                     // we have already tried this request on the write location
+                    logger.warn("SessionNotAvailable: canUseMultipleWriteLocations = false, sessionTokenRetryCount = {}, no more retries", sessionTokenRetryCount);
```

Comment:
If you send another iteration, can we please use variable itself in logger for canUseMultipleWriteLocations , instead of true and false

Reply:
what's the advantage? in the if branch the value is deterministic

Reply:
Good point @simplynaveen20

Reply:
intellj gives a recommendation in general to use
```java
if (flag) {
  func(true)
}
```
instead of
```java
if (flag) {
  func(flag)
}
```
that's why I am not using a variable here.